File: 0161-one-edit-distance/0161-one-edit-distance.cpp
class Solution {
public:
    bool isOneEditDistance(string s, string t) {
        if(s.empty() && t.empty()) return false; 
        int len = min(s.length(), t.length()); 

        for(int i = 0; i < len; i++){
            if(s[i] != t[i]){
                if(s.length() == t.length()){
                    return (s.substr(i + 1) == t.substr(i + 1)); 
                } 
                if(s.length() < t.length()){
                    return (t.substr(i+1) == s.substr(i)); 
                }
                if(s.length() > t.length()){
                    return (s.substr(i+1) == t.substr(i)); 
                }
            }
        }
        if(s.length() > t.length()){
            return (t.length() + 1 == s.length()); //insert 
        }
        return (s.length() + 1 == t.length()); //insert
    }
};

File: 0161-one-edit-distance/0161-one-edit-distance.java
class Solution {
    public boolean isOneEditDistance(String s, String t) {
        if(s.equals(t)) return false; 
        if(s.length() > t.length()){
            String tmp = s; 
            s = t; 
            t = tmp; 
        }

        for(int i = 0; i < s.length(); i++){
            char sChar = s.charAt(i); 
            char tChar = t.charAt(i); 

            if(sChar != tChar){
                if(s.length() == t.length()){
                    return s.substring(i+1).equals(t.substring(i+1)); 
                } 
                if(s.length() < t.length()){
                    return s.substring(i).equals(t.substring(i+1)); 
                }
            }
        }

        return s.length()+1 == t.length(); 
    }
}